examples/adder/tests/test_adder.py

```diff
@@ -1,47 +1,38 @@
 # Simple tests for an adder module
 import cocotb
 from cocotb.triggers import Timer
-from cocotb.result import TestFailure
 from adder_model import adder_model
 import random
 
 
 @cocotb.test()
-def adder_basic_test(dut):
+async def adder_basic_test(dut):
     """Test for 5 + 10"""
-    yield Timer(2, units='ns')
+
     A = 5
     B = 10
 
-    dut.A = A
-    dut.B = B
+    dut.A <= A
+    dut.B <= B
 
-    yield Timer(2, units='ns')
+    await Timer(2, units='ns')
 
-    if int(dut.X) != adder_model(A, B):
-        raise TestFailure(
-            "Adder result is incorrect: %s != 15" % str(dut.X))
-    else:  # these last two lines are not strictly necessary
-        dut._log.info("Ok!")
+    assert int(dut.X) == adder_model(A, B), "Adder result is incorrect: {} != 15".format(dut.X)
 
 
 @cocotb.test()
-def adder_randomised_test(dut):
+async def adder_randomised_test(dut):
     """Test for adding 2 random numbers multiple times"""
-    yield Timer(2, units='ns')
 
     for i in range(10):
+
         A = random.randint(0, 15)
         B = random.randint(0, 15)
 
-        dut.A = A
-        dut.B = B
+        dut.A <= A
+        dut.B <= B
```

Comment:
Prefer this syntax. I've seen at least twice this week people thinking the following should work based on `__setattr__` working. We shouldn't confuse people. Alternatively, we could add support for overloading the assignment operator, which I've mentioned before.

```python
alias = dut.signal
alias = 4
# why isn't dut.signal 4?
```